File: src/main/java/be/hubau/sane/parser/Terminal_NumericValue.java
package be.hubau.sane.parser;/* -----------------------------------------------------------------------------
 * Terminal_NumericValue.java
 * -----------------------------------------------------------------------------
 *
 * Producer : com.parse2.aparse.Parser 2.5
 * Produced : Mon Apr 15 15:26:09 CEST 2019
 *
 * -----------------------------------------------------------------------------
 */

import java.util.ArrayList;
import java.util.regex.Pattern;

public class Terminal_NumericValue extends Rule {
    private Terminal_NumericValue(String spelling, ArrayList<Rule> rules) {
        super(spelling, rules);
    }

    public static Terminal_NumericValue parse(
            ParserContext context,
            String spelling,
            String regex,
            int length) {
        context.push("NumericValue", spelling + "," + regex);

        boolean parsed = true;

        Terminal_NumericValue numericValue = null;
        try {
            String value =
                    context.text.substring(
                            context.index,
                            context.index + length);

            if ((parsed = Pattern.matches(regex, value))) {
                context.index += length;
                numericValue = new Terminal_NumericValue(value, null);
            }
        } catch (IndexOutOfBoundsException e) {
            parsed = false;
        }

        context.pop("NumericValue", parsed);

        return numericValue;
    }

    public Object accept(Visitor visitor) {
        return visitor.visit(this);
    }
}
/* -----------------------------------------------------------------------------
 * eof
 * -----------------------------------------------------------------------------
 */

File: src/main/java/be/hubau/sane/parser/TerminalNumericValue.kt
package be.hubau.sane.parser

import java.util.*
import java.util.regex.Pattern

class TerminalNumericValue private constructor(spelling: String, rules: ArrayList<Rule>?) : Rule(spelling, rules) {

    override fun accept(visitor: Visitor): Any? {
        return visitor.visit(this)
    }

    companion object {

        fun parse(
                context: ParserContext,
                spelling: String,
                regex: String,
                length: Int): TerminalNumericValue? {
            context.push("NumericValue", "$spelling,$regex")

            var parsed: Boolean

            var numericValue: TerminalNumericValue? = null
            try {
                val value = context.text.substring(context.index, context.index + length)

                parsed = Pattern.matches(regex, value)
                if (parsed) {
                    context.index += length
                    numericValue = TerminalNumericValue(value, null)
                }
            } catch (e: IndexOutOfBoundsException) {
                parsed = false
            }

            context.pop("NumericValue", parsed)

            return numericValue
        }
    }
}